src/agent/onefuzz-supervisor/src/worker.rs

```diff
@@ -195,7 +202,12 @@ impl IWorkerRunner for WorkerRunner {
 
         let config_path = work.config_path()?;
 
-        fs::write(&config_path, work.config.expose_ref()).await?;
+        // injecting the setup_dir path in the agent config
+        let mut json_config: serde_json::Value = serde_json::from_str(work.config.expose_ref())?;
```

Comment:
Do we really want this to be part of the task config?

I feel like it would be better for the supervisor to pass the container information that it manages to the worker agent when invoking, and then the agent would then use that container info in its context.

Reply:
So, even if we don't want to do a full "supervisor-managed container registry" right now, would it still be better to pass just the `setup_dir` explicitly to the agent binary?

Reply:
how would you suggest that we pass the setup_dir to the individual tasks without modifying the task config? I get the idea of sending the info from the supervisor_agent to the worker_agent via a parameter. But that info needs to eventually get to the tasks from the config.

I am ok to switch to using a paramter to use a paramter to pass the setup_dir but it will sill be part of the config.